lib/ProMotion/collection/collection.rb

```diff
@@ -0,0 +1,106 @@
+module ProMotion
+  module Collection
+    include ProMotion::Styling
+    include ProMotion::Table::Utils
+    include ProMotion::CollectionBuilder
+
+    attr_reader :promotion_collection_data
+
+    def collection_view
+      self.collectionView
+    end
+
+    def screen_setup
+      check_collection_data
+    end
+
+    # Override viewDidLoad here to register cell classes here.
+    # Calling it from `screen_setup` cause `on_load` to be called twice
```

Comment:
From what I've been able to find/trace, it looks like that initialize self.collectionView too early cause this viewDidLoad call